spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/PropertiesRedisConnectionDetails.java

```diff
@@ -113,8 +113,10 @@ public Cluster getCluster() {
 	}
 
 	private Node asNode(String node) {
-		String[] components = node.split(":");
-		return new Node(components[0], Integer.parseInt(components[1]));
+		var posLastColon = node.lastIndexOf(':');
+		var host = node.substring(0, posLastColon).replaceFirst("^\\[(.+)]$", "$1");
```

Comment:
I don't think the `[` and `]` should be removed. They can be useful in situations where the `:` characters in an address can cause problems. We should trust the user to have used them in the property's value intentionally.

I think we can just use code that's very similar to `PropertiesRabbitConnectionDetails`:

https://github.com/spring-projects/spring-boot/blob/9a4d0fb8b62aac303e9879f625cf4e7a4cf43cbc/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/PropertiesRabbitConnectionDetails.java#L56-L59